numpy/core/tests/test_multiarray.py

```diff
@@ -4541,6 +4541,16 @@ def test_file_position_after_tofile(self):
             f.close()
             assert_equal(pos, 10, err_msg=err_msg)
 
+    def test_allocation_error_fromfile(self):
+        # gh-12300
+        with open(self.filename, 'wb') as f:
+            f.write(b'\xfe')
+
+        with open(self.filename, 'rb') as f:
+            assert_raises(ValueError, np.fromfile, f, np.unicode)
+
+        assert_raises(ValueError, np.fromfile, self.filename, np.unicode)
```

Comment:
This test isn't quite the same as the case reported in the issue.

The reason here is that on some platforms an overcommitted allocation could be possible, by attempting a read of invalid unicode data we get a test which is not flaky.